mantidproject/mantid
# Fix updating Matrix display when the workspace is replaced

Depends on ~~#24700~~. This issue was discovered in #24595 and fixes a bug there.

**Description of work.**
Matrix display is now properly updated when the workspace is replaced. Previously it would not properly replace the workspace reference in the QTableView models, thus not updating when the workspace pointer is replaced.

**To test:**
- Load a workspace.
- Show its data
- Multiply it by itself - data should change
- Reload the original workspace - data should be restored to original 

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
